src/client/src/error.rs

```diff
@@ -1,28 +1,27 @@
 use std::io::Error as IoError;
-use thiserror::Error;
 
-use fluvio_socket::FlvSocketError;
+use fluvio_socket::SocketError;
 use fluvio_sc_schema::ApiError;
 use crate::config::ConfigError;
 use semver::Version;
 
 /// Possible errors that may arise when using Fluvio
-#[derive(Error, Debug)]
+#[derive(thiserror::Error, Debug)]
 pub enum FluvioError {
+    #[error(transparent)]
+    Io(#[from] IoError),
     #[error("Topic not found: {0}")]
     TopicNotFound(String),
     #[error("Partition not found: {0}-{1}")]
     PartitionNotFound(String, i32),
     #[error("Spu not found: {0}")]
     SPUNotFound(i32),
-    #[error(transparent)]
-    IoError(#[from] IoError),
     #[error("Fluvio socket error")]
-    FlvSocketError(#[from] FlvSocketError),
+    Socket(#[from] SocketError),
     #[error("Fluvio SC schema error")]
-    ApiError(#[from] ApiError),
+    ScSchema(#[from] ApiError),
     #[error("Fluvio config error")]
-    ConfigError(#[from] ConfigError),
+    Config(#[from] ConfigError),
```

Comment:
this is too generic. Change this to ProfileConfig

Reply:
What about `ClientConfig`?